### PR TITLE
Create root pom.xml

### PR DESCRIPTION
**Pull request by TimvdLippe**
This PR introduces a root `pom.xml` to be able to import this project more easily in an editor. Moreover, building the project is now a simple `mvn install` in the root folder, instead of running the Travis bash file.

Sadly the number of files changed is largely, but a summary of the changes is:
1. Create root `pom.xml` to replace custom directory changing logic from `travis_build.sh`
2. Move `WatchDogIntellijPlugin/WatchDog` to `WatchDogIntellijPlugin`. This interim directory was not necessary and was blocking step 3.
3. Introduce `maven-exec-plugin` with maven lifecycle mappings in order to succesfully run all tests and fetch the prerequisites. This means that `mvn test` will now properly run the Ruby tests in `WatchDogServer` and `mvn install` will properly fetch a local version of IntelliJ if necessary
4. I have also updated the IntelliJ configuration files to load in my editor.
5. Ignore all Eclipse `.settings/` which seemed to clutter the diff a lot. At the same time the `.gitignore` has been updated to reduce duplication and only ignore the files that are cluttering the diffs.

The introduction of a root `pom.xml` also meant that I had to update the various artifact ID's of the subprojects, as there were duplicate IDs. Therefore most of the changes reflect these renames, as well as in the actual code (magic Strings FTW).

### CODE REVIEW

**Comment by TimvdLippe** [comment]
Investigating the issue of installing the Intellij local dependency right now.

**Comment by TimvdLippe** [comment]
@Inventitech All right, this PR is now up for review. I made some typos and wrong assumptions with Travis building. Note that the `push` build failed because of flaky tests, but `pr` build is working just fine. Sadly I can't retrigger Travis builds, but it should "just" work after restarting that failed job.

File: .idea/compiler.xml [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+        <module name="nl.tudelft.watchdog.p2updatesite" />
+        <module name="nl.tudelft.WatchDogCore" />
+        <module name="WatchDog" />
+        <module name="watchdog.server" />
+        <module name="WatchDogCore" />
+        <module name="WatchDogUnitTests" />
+      </profile>
+    </annotationProcessing>
+    <bytecodeTargetLevel>
+      <module name="nl.tudelft.WatchDog.eclipse" target="1.5" />
```
Comment (Inventitech) [inline .idea/compiler.xml]: Convert everything to 1.8?

File: .idea/dictionaries/tim.xml [hunk 2]
```diff
@@ -0,0 +1,3 @@
+<component name="ProjectDictionaryState">
+  <dictionary name="tim" />
```
Comment (Inventitech) [inline .idea/dictionaries/tim.xml]: Candidate for gitignore?

File: .idea/libraries/Maven__joda_time_joda_time_2_8_1.xml [hunk 3]
```diff
@@ -0,0 +1,13 @@
+<component name="libraryTable">
```
Comment (Inventitech) [inline .idea/libraries/Maven__joda_time_joda_time_2_8_1.xml]: Get rid of this file?

File: WatchDogCore/WatchDogCoreFeature/WatchDogCore.iml [hunk 4]
```diff
@@ -1,6 +1,6 @@
 <?xml version="1.0" encoding="UTF-8"?>
```
Comment (Inventitech) [inline WatchDogCore/WatchDogCoreFeature/WatchDogCore.iml]: Maybe able to remove?

File: WatchDogCore/WatchDogCoreFeature/pom.xml [hunk 5]
```diff
@@ -1,15 +1,42 @@
...
+		<version>2.0.0</version>
+	</parent>
 
-  <artifactId>WatchDogCore</artifactId>
-  <packaging>eclipse-feature</packaging>
+	<artifactId>WatchDogCore</artifactId>
+	<packaging>eclipse-feature</packaging>
 
-</project> 
+	<build>
```
Comment (Inventitech) [inline WatchDogCore/WatchDogCoreFeature/pom.xml]: Probably can get rid of this whole block?

File: WatchDogEclipsePlugin/WatchDog/.settings/org.eclipse.jdt.core.prefs [hunk 6]
```diff
@@ -1,316 +0,0 @@
...
-org.eclipse.jdt.core.codeComplete.fieldSuffixes=
-org.eclipse.jdt.core.codeComplete.localPrefixes=
-org.eclipse.jdt.core.codeComplete.localSuffixes=
-org.eclipse.jdt.core.codeComplete.staticFieldPrefixes=
-org.eclipse.jdt.core.codeComplete.staticFieldSuffixes=
-org.eclipse.jdt.core.codeComplete.staticFinalFieldPrefixes=
-org.eclipse.jdt.core.codeComplete.staticFinalFieldSuffixes=
-org.eclipse.jdt.core.compiler.codegen.inlineJsrBytecode=enabled
-org.eclipse.jdt.core.compiler.codegen.methodParameters=do not generate
-org.eclipse.jdt.core.compiler.codegen.targetPlatform=1.7
```
Comment (Inventitech) [inline WatchDogEclipsePlugin/WatchDog/.settings/org.eclipse.jdt.core.prefs]: Change to 1.8

File: WatchDogEclipsePlugin/WatchDog/.settings/org.eclipse.jdt.core.prefs [hunk 7]
```diff
@@ -1,316 +0,0 @@
...
-org.eclipse.jdt.core.compiler.codegen.unusedLocal=preserve
-org.eclipse.jdt.core.compiler.compliance=1.7
```
Comment (Inventitech) [inline WatchDogEclipsePlugin/WatchDog/.settings/org.eclipse.jdt.core.prefs]: Change to 1.8

File: WatchDogEclipsePlugin/WatchDogUnitTests/META-INF/MANIFEST.MF [hunk 8]
```diff
@@ -3,7 +3,7 @@ Bundle-ManifestVersion: 2
 Bundle-Name: WatchDogUnitTests
 Bundle-SymbolicName: WatchDogUnitTests
 Bundle-Version: 2.0.0
-Fragment-Host: nl.tudelft.WatchDog;bundle-version="[0.0.0,3.0.0)"
+Fragment-Host: nl.tudelft.WatchDog.eclipse;bundle-version="[0.0.0,3.0.0)"
 Bundle-RequiredExecutionEnvironment: JavaSE-1.7
```
Comment (Inventitech) [inline WatchDogEclipsePlugin/WatchDogUnitTests/META-INF/MANIFEST.MF]: 1.8 ;)

File: WatchDogIntelliJPlugin/src/nl/tudelft/watchdog/intellij/logic/InitializationManager.java [hunk 9]
```diff
@@ -51,7 +51,7 @@
     private InitializationManager(Project project) {
```
Comment (Inventitech) [inline WatchDogIntelliJPlugin/src/nl/tudelft/watchdog/intellij/logic/InitializationManager.java]: This is likely a problem for accessing the local settings + database of legacy users

**Comment by Inventitech** [comment]
Have a look at the reactor summary, too:
Now:
```
[INFO] Reactor Summary:

[INFO] 

[INFO] parent ............................................. SUCCESS [  0.005 s]

[INFO] watchdog-core-parent ............................... SUCCESS [  0.001 s]

[INFO] nl.tudelft.WatchDogCore ............................ SUCCESS [  0.985 s]

[INFO] WatchDogCore ....................................... SUCCESS [  0.087 s]

[INFO] watchdog-eclipse-plugin ............................ SUCCESS [  0.003 s]

[INFO] nl.tudelft.WatchDog.eclipse ........................ SUCCESS [ 14.594 s]

[INFO] WatchDogUnitTests .................................. SUCCESS [ 30.341 s]

[INFO] WatchDog ........................................... SUCCESS [  0.094 s]

[INFO] nl.tudelft.watchdog.p2updatesite ................... SUCCESS [ 18.702 s]

[INFO] watchdog-intellij-plugin ........................... SUCCESS [  1.089 s]

[INFO] watchdog.server .................................... SUCCESS [  2.112 s]
```

Used to be:
```
[INFO] Reactor Summary:

[INFO] 

[INFO] parent ............................................. SUCCESS [  0.003 s]

[INFO] nl.tudelft.WatchDog ................................ SUCCESS [ 23.714 s]

[INFO] WatchDogUnitTests .................................. SUCCESS [ 35.019 s]

[INFO] WatchDog ........................................... SUCCESS [  0.102 s]

[INFO] nl.tudelft.watchdog.p2updatesite ................... SUCCESS [ 20.313 s]

[INFO] ------------------------------------------------------------------------
```

The number of built entities and their naming schema is quite confusing now.

**Comment by TimvdLippe** [comment]
RE: Reactor

Yes that is different, as the root pom.xml now executes every single module. Most modules have no tests or anything, so that is not needed. I was planning on addressing the naming schema in a subsequent PR, to not clutter this one.

File: WatchDogCore/WatchDogCoreFeature/WatchDogCore.iml (hunk 4 above)
Comment (TimvdLippe) [reply WatchDogCore/WatchDogCoreFeature/WatchDogCore.iml]: Sadly no :cry:

File: WatchDogEclipsePlugin/WatchDog/.settings/org.eclipse.jdt.core.prefs (hunk 6 above)
Comment (TimvdLippe) [reply WatchDogEclipsePlugin/WatchDog/.settings/org.eclipse.jdt.core.prefs]: This should be auto-generated now correctly. If you import the project fresh, they should be there.

**Comment by TimvdLippe** [comment]
@Inventitech I have added a CONTRIBUTING.md (https://github.com/TestRoots/watchdog/blob/parent-pom/CONTRIBUTING.md) with the updated instructions to import this project. Especially the IntelliJ instructions should be a lot easier now :smile:

File: WatchDogIntelliJPlugin/src/nl/tudelft/watchdog/intellij/logic/InitializationManager.java (hunk 9 above)
Comment (TimvdLippe) [reply WatchDogIntelliJPlugin/src/nl/tudelft/watchdog/intellij/logic/InitializationManager.java]: I have created a forum post to ask this: https://intellij-support.jetbrains.com/hc/en-us/community/posts/115000777330-Changing-plugin-ID-with-update

**Pull request merged**